src/components/TokenSelector.tsx

```diff
@@ -262,16 +288,15 @@ const TokenSelector: React.FC<Props> = ({ isDisabled, tokens, selected, onChange
         classNamePrefix="react-select"
         noOptionsMessage={(): string => 'No results'}
         formatOptionLabel={formatOptionLabel}
-        // menuIsOpen={true}  // uncomment to be able to interact with menu styles
         options={options}
         value={{ token: selected }}
-        onChange={(selected: { token: TokenDetails }, { action }: ActionMeta): void => {
-          if (selected && action === 'select-option' && 'token' in selected) {
-            onChange(selected.token)
-          }
-        }}
+        onChange={onSelectChange}
         tabIndex={tabIndex.toString()}
-        components={{ MenuList: menuListFactory() }}
+        components={{ MenuList }}
```

Comment:
Since we have this statically, no need to recreate components object on each render
Do `const component ={}` outside of TokenSelector
Same for `noOptionsMessage`